internal/tofu/node_resource_plan_orphan.go

```diff
@@ -54,22 +58,44 @@ func (n *NodePlannableResourceInstanceOrphan) Name() string {
 }
 
 // GraphNodeExecutable
-func (n *NodePlannableResourceInstanceOrphan) Execute(_ context.Context, evalCtx EvalContext, op walkOperation) tfdiags.Diagnostics {
+func (n *NodePlannableResourceInstanceOrphan) Execute(ctx context.Context, evalCtx EvalContext, op walkOperation) tfdiags.Diagnostics {
 	addr := n.ResourceInstanceAddr()
 
+	ctx, span := tracing.Tracer().Start(
+		ctx, traceNamePlanResourceInstance,
+		otelTrace.WithAttributes(
+			otelAttr.String(traceAttrResourceInstanceAddr, addr.String()),
+			otelAttr.Bool(traceAttrPlanRefresh, !n.skipRefresh),
+			otelAttr.Bool(traceAttrPlanPlanChanges, !n.skipPlanChanges),
+		),
+	)
+	defer span.End()
+
 	// Eval info is different depending on what kind of resource this is
+	var diags tfdiags.Diagnostics
 	switch addr.Resource.Resource.Mode {
 	case addrs.ManagedResourceMode:
-		diags := n.resolveProvider(evalCtx, true, states.NotDeposed)
-		if diags.HasErrors() {
+		resolveDiags := n.resolveProvider(evalCtx, true, states.NotDeposed)
+		diags = diags.Append(resolveDiags)
+		if resolveDiags.HasErrors() {
```

Comment:
`suggestion`
```suggestion
		if diags.HasErrors() {
```
It is just confusing here... I understand that `diags` is just created above, but checking `resolveDiags` for errors and using `diags` afterwards is just confusing.
Feel free to ignore if you had a good argument for doing it this way.

Reply:
This pattern is quite common in the codebase already because in the case where the diagnostics contains warnings (and so we don't return immediately) a subsequent `diags.HasErrors()` would need to re-check whether each warning is an error.

Checking `resolveDiags` here means something like "are any of the _new_ diagnostics representing errors?", since we typically already know from an earlier check that nothing that was already in `diags` was an error.